pipeline.py

```diff
@@ -72,31 +91,100 @@ class TextProcessingPipeline:
     """
 
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+        articles = self.corpus_manager.get_articles()
+
+        for article in articles.values():
+            article_text = article.get_raw_text()
+            cleaned_tokens = []
+            single_tagged_tokens = []
+            multiple_tagged_tokens = []
+
+            for token in self._process(article_text):
+                cleaned_tokens.append(token.get_cleaned())
+                single_tagged_tokens.append(token.get_single_tagged())
+                multiple_tagged_tokens.append(token.get_multiple_tagged())
+
+            article.save_as(' '.join(cleaned_tokens), ArtifactType.cleaned)
+            article.save_as(' '.join(single_tagged_tokens), ArtifactType.single_tagged)
+            article.save_as(' '.join(multiple_tagged_tokens), ArtifactType.multiple_tagged)
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        normal_text = raw_text.replace("\n", " ")
+
+        analyzed_text = Mystem().analyze(normal_text)
+        morphs = pymorphy2.MorphAnalyzer()
+
+        tokens = []
+        for token in analyzed_text:
+            if not token.get("analysis"):
+                continue
+            if "analysis" not in token:
+                continue
+            morph_token = MorphologicalToken(original_word=token['text'])
+            morph_token.normalized_form = token['analysis'][0]['lex']
+            morph_token.tags_mystem = token['analysis'][0]['gr']
+            morph_token.tags_pymorphy = morphs.parse(token['text'])[0].tag
+            tokens.append(morph_token)
+        return tokens
 
 
 def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
+    if isinstance(path_to_validate, str):
+        path_to_validate = Path(path_to_validate)
+
+    if not path_to_validate.exists():
+        raise FileNotFoundError
+
+    if not path_to_validate.is_dir():
+        raise NotADirectoryError
+
+    data = list(path_to_validate.glob('*'))
+
+    if not data:
+        raise EmptyDirectoryError
+
+    number_of_texts = 0
+    number_of_metas = 0
+
+    pattern = re.compile(r'[0-9]+')
+    for file in sorted(list(path_to_validate.glob('*')), key=lambda x: int(pattern.search(x.name).group(0))):
+        if "_raw.txt" in file.name:
+            with open(file, 'r', encoding='utf-8') as text_file:
```

Comment:
should be a better way to understand whether a file is empty. ask colleagues if you get stuck